packages/components/src/spectrum/CheckboxGroup.tsx

```diff
@@ -0,0 +1,61 @@
+import { isElementOfType } from '@deephaven/react-hooks';
+import React, { ReactNode, useMemo, useState } from 'react';
+import {
+  Checkbox,
+  CheckboxGroup as SpectrumCheckboxGroup,
+  SpectrumCheckboxGroupProps,
+} from '@adobe/react-spectrum';
+
+export type CheckboxGroupProps = {
+  children: ReactNode;
+} & Omit<SpectrumCheckboxGroupProps, 'children'>;
+
+/**
+ * Augmented version of the Spectrum CheckboxGroup component that supports
+ * primitive item children.
+ */
+export function CheckboxGroup({
+  children,
+  ...props
+}: CheckboxGroupProps): JSX.Element {
+  const [checkedState, setCheckedState] = useState<{ [key: number]: boolean }>(
+    {}
+  );
+
+  const handleCheckboxChange = (index: number) => {
+    setCheckedState(prevState => ({
+      ...prevState,
+      [index]: !prevState[index],
+    }));
+  };
+
+  const wrappedChildren = useMemo(
+    () =>
+      React.Children.map(children, (child, index) => {
+        if (isElementOfType(child, Checkbox)) {
+          return React.cloneElement(child, {
```

Comment:
There shouldn't be a need to clone or set any props if a Checkbox was provided. In cases where the primitive is provided, should only have to set value and children props. The CheckboxGroup does the rest